src/main/java/org/gridsuite/odre/server/client/OdreOpenDataClientImpl.java

```diff
@@ -0,0 +1,88 @@
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.web.client.RestTemplateBuilder;
+import org.springframework.http.*;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.DefaultUriBuilderFactory;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import java.io.*;
+
+import java.util.*;
+
+/**
+ * Parse RTE substation and line segment coordinates.
+ * <ul>
+ *     <li>
+ *         <a href="https://opendata.reseaux-energies.fr/explore/dataset/postes-electriques-rte/download/?format=csv&timezone=Europe/Berlin&use_labels_for_header=true">postes-electriques-rte.csv</a>
+ *     </li>
+ *     <li>
+ *         <a href="https://opendata.reseaux-energies.fr/explore/dataset/lignes-aeriennes-rte/download/?format=csv&timezone=Europe/Berlin&use_labels_for_header=true">lignes-aeriennes-rte.csv</a>
+ *     </li>
+ *     <li>
+ *         <a href="https://opendata.reseaux-energies.fr/explore/dataset/lignes-souterraines-rte/download/?format=csv&timezone=Europe/Berlin&use_labels_for_header=true">lignes-souterraines-rte.csv</a>
+ *     </li>
+ * </ul>
+ *
+ * @author Chamseddine Benhamed <chamseddine.benhamed at rte-france.com>
+ */
+
+@Component
+public class OdreOpenDataClientImpl implements OdreClient {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(OdreOpenDataClientImpl.class);
+
+    private final String openDataBaseUri = "https://opendata.reseaux-energies.fr";
+
+    private RestTemplate openDataRest;
+
+    @Autowired
+    public OdreOpenDataClientImpl() {
+        RestTemplateBuilder restTemplateBuilder = new RestTemplateBuilder();
+        this.openDataRest = restTemplateBuilder.build();
+        this.openDataRest.setUriTemplateHandler(new DefaultUriBuilderFactory(openDataBaseUri));
+    }
+
+    private ByteArrayInputStream downloadFile(String path) {
+        String uri = UriComponentsBuilder.fromPath(path)
+                .queryParam("format", "csv")
+                .toUriString();
+
+        ResponseEntity<byte[]> responseEntity = openDataRest.exchange(uri,
+                HttpMethod.GET,
+                HttpEntity.EMPTY,
+                byte[].class);
+
+        return new ByteArrayInputStream(Objects.requireNonNull(responseEntity.getBody()));
+    }
+
+    @Override
+    public List<SubstationGeoData> getSubstations() {
+        ByteArrayInputStream byteArrayInputStream = downloadFile("/explore/dataset/postes-electriques-rte/download/");
+        LOGGER.info("substations were downloaded from the open data server");
+        return new ArrayList<>(GeographicDataParser.parseSubstations(new BufferedReader(new InputStreamReader(byteArrayInputStream))).values());
+    }
+
+    @Override
+    public List<LineGeoData> getLines() {
+        ByteArrayInputStream undergroundLinesByteArrayInputStream =  downloadFile("/explore/dataset/lignes-souterraines-rte/download/");
```

Comment:
Is it better to make these URLs parametrable or we don't have any use case that requires it ?

Reply:
we don't have any use case that requires it